src/chartiq_injections/findHighlights.js

```diff
@@ -0,0 +1,543 @@
+                    pixelBox.y1,
+                    pixelPoint1.x0,
+                    pixelPoint1.y0,
+                    pixelPoint1.x1,
+                    pixelPoint1.y1,
+                    'segment'
+                ) ||
+                CIQ.boxIntersects(
+                    pixelBox.x0,
+                    pixelBox.y0,
+                    pixelBox.x1,
+                    pixelBox.y1,
+                    pixelPoint2.x0,
+                    pixelPoint2.y0,
+                    pixelPoint2.x1,
+                    pixelPoint2.y1,
+                    'segment'
+                )
+            ) {
+                return true;
+            }
+            return false;
+        }
+
+        if (!clearOnly && !this.anyHighlighted && this.controls.anchorHandles) {
+            for (const id in this.controls.anchorHandles) {
+                const anchorHandle = this.controls.anchorHandles[id];
+                const { handle, sd } = anchorHandle;
+                const xo = this.resolveX(cx);
+                const yo = this.resolveY(cy);
+                if (handle) {
+                    const { left, top, right, bottom } = handle.getBoundingClientRect();
+                    if (CIQ.boxIntersects(left, top, right, bottom, xo, yo, xo, yo)) {
+                        anchorHandle.highlighted = true;
+                        this.anyHighlighted = true;
+                        somethingChanged = true;
+                        stickyArgs = {
+                            message: sd.name,
+                            type: 'anchorHandle',
+                        };
+                        continue;
+                    }
+
+                    if (anchorHandle.highlighted === true) {
+                        anchorHandle.highlighted = false;
+                        somethingChanged = true;
+                    }
+                }
+            }
+        }
+
+        if (!clearOnly && !this.anyHighlighted && chart.dataSegment) {
+            bar = this.barFromPixel(cx);
+            if (bar >= 0 && bar < chart.dataSegment.length) {
+                let y;
+                for (n in this.overlays) {
+                    o = this.overlays[n];
+
+                    // check handles before this to make sure to set highlight state to false where appropriate
+                    if (o.panel !== this.currentPanel.name) continue;
+
+                    // custom highlight detection
+                    if (o.study.isHighlighted === false) continue;
+                    else if (typeof o.study.isHighlighted === 'function') {
+                        if (o.study.isHighlighted(this, cx, cy)) {
+                            o.highlight = true;
+                            this.anyHighlighted = true;
+                        }
+                        continue;
+                    }
+
+                    const quote = chart.dataSegment[bar];
+                    if (!quote) continue;
+                    let out;
```

Comment:
Note: had to replace all vars with `let`s & `const`s that's why I introduce this `out` variable here instead of declaring `var out` at line 329 for the first time.